tools/generators/pbxproject_prefix/test/CalculateCompatibilityVersionTests.swift

```diff
@@ -0,0 +1,103 @@
+import GeneratorCommon
+import XCTest
+
+@testable import pbxproject_prefix
+
+class CalculateCompatibilityVersionTests: XCTestCase {
+    func test_lowXcodeVersion() {
+        // Arrange
+
+        let minimumXcodeVersion: SemanticVersion = "11.4.7"
+        let expectedCompatibilityVersion = "Xcode 11.0"
+
+        // Act
+
+        let compatibilityVersion = Generator.calculateCompatibilityVersion(
+            minimumXcodeVersion: minimumXcodeVersion
+        )
+
+        // Assert
+
+        XCTAssertEqual(
+            compatibilityVersion,
+            expectedCompatibilityVersion
+        )
+    }
+
+    func test_xcode13() {
+        // Arrange
+
+        let minimumXcodeVersion: SemanticVersion = "13.4.2"
+        let expectedCompatibilityVersion = "Xcode 13.0"
+
+        // Act
+
+        let compatibilityVersion = Generator.calculateCompatibilityVersion(
+            minimumXcodeVersion: minimumXcodeVersion
+        )
+
+        // Assert
+
+        XCTAssertEqual(
+            compatibilityVersion,
+            expectedCompatibilityVersion
+        )
+    }
+
+    func test_xcode14() {
+        // Arrange
+
+        let minimumXcodeVersion: SemanticVersion = "14.2.1"
```

Comment:
What if a user pass `14.2.0.14C18`? Is this something we should handle/error out nicely if it's not expected?

Reply:
That will never be sent to this tool. We force a specific format for this version here: https://github.com/MobileNativeFoundation/rules_xcodeproj/blob/925d111ccfb8d65f7701e352ae6daeddf9e61b96/xcodeproj/internal/xcodeproj_rule.bzl#L122-L129

Reply:
👍